src/detect_weeds.py

```diff
@@ -69,11 +69,15 @@ def detect_weeds(self, image: np.ndarray) -> Optional[Dict[str, Dict[str, int]]]
         log.info("Starting weed detection.")
         results = self.model(image)
 
-        if not results or not results[0].boxes.xyxy.tolist():
-            log.warning("No detection found.")
+        try:        
```

Comment:
keep the previous version just add the image path in the log warning message so we can inspect